### PR TITLE
block user from booking multiple times a day

### PR DESCRIPTION
**Pull request by jimmy-cyr**
block user from booking multiple times a day

### CODE REVIEW

**Review by serherca — APPROVED** [review]

File: Book-A-Desk.Domain/Reservation.Validator.fs [hunk 1]
```diff
@@ -49,12 +49,24 @@ module BookADeskReservationValidator =
         else
             return! Error ($"The office is booked out at {date.ToShortDateString()}" )
     }
+    
+    let private validateUserHasNotBookedYet reservationAggregate emailAddress officeId (date : DateTime) = result {
+        let alreadyBooked =
+            reservationAggregate.BookedDesks
+            |> List.filter(fun b -> b.Date.Date = date.Date && b.OfficeId = officeId && b.EmailAddress = emailAddress)
```
Comment (AlexGagne) [inline Book-A-Desk.Domain/Reservation.Validator.fs:56]: Rename b to bookedDesk

File: Book-A-Desk.Domain/Reservation.Validator.fs [hunk 2]
```diff
@@ -49,12 +49,24 @@ module BookADeskReservationValidator =
         else
             return! Error ($"The office is booked out at {date.ToShortDateString()}" )
     }
+    
+    let private validateUserHasNotBookedYet reservationAggregate emailAddress officeId (date : DateTime) = result {
+        let alreadyBooked =
```
Comment (AlexGagne) [inline Book-A-Desk.Domain/Reservation.Validator.fs:54]: alreadyBookedDesks

File: Book-A-desk.Domain.Tests/Reservation.Validator.Tests.fs [hunk 3]
```diff
@@ -123,4 +123,43 @@ let ``GIVEN A valid Book-A-Desk Reservation command, WHEN validating the command
...
+    let result = BookADeskReservationValidator.validateCommand offices command aReservationAggregate
+    match result with
+    | Error _ -> failwith "Validation should have succeeded"
+    | Ok _ -> ()
+    
+[<Fact>]
+let ``GIVEN A Book-A-Desk Reservation command with the user already booked, WHEN validating the command, THEN validation should fail.`` () =
+    let command =
+        {
+            EmailAddress = EmailAddress "anEmailAddress@fake.com"
```
Comment (AlexGagne) [inline Book-A-desk.Domain.Tests/Reservation.Validator.Tests.fs:146]: Extract the email to its own variable and reuse it in the reservation aggregate and command

File: Book-A-desk.Domain.Tests/Reservation.Validator.Tests.fs [hunk 4]
```diff
@@ -123,4 +123,43 @@ let ``GIVEN A valid Book-A-Desk Reservation command, WHEN validating the command
...
+            Date = DateTime.MaxValue
+            OfficeId = office.Id
+        } : BookADesk
+    
+    let aReservationAggregate =
+        {
+            Id = ReservationAggregate.Id
+            BookedDesks = [
+                                { A.booking with
```
Comment (AlexGagne) [inline Book-A-desk.Domain.Tests/Reservation.Validator.Tests.fs:155]: Make sure you explicitely specify the booking date on this + command

**Pull request merged**